### DISCUSSION TITLE
Add own ssl certificate?

### DISCUSSION THREAD

**Discussion by ConiKost** [john30/ebusd#1175] [Q&A] [answered]
Is there any way, if https is enabled, to add an own ssl certificate instead of the self-signed one? I don't see any option within "easi>", or did i missed it?

**Comment by Johnadam23** [comment]
Here are some typical actions you can take in case you want to change the self-signed SSL certificate in a system or application named "easi>" with your own SSL certificate but can't find an option for it in the interface:

**Find SSL Certificate Directory:** To begin with, locate the directory in the "easi>" system where the SSL certificates are kept. Usually, the self-signed certificate is kept in this directory.

**Create or Get an SSL Certificate:** [Get an SSL certificate](https://www.ssl2buy.com/) from a reliable Certificate Authority (CA) or, if you'd rather, create a self-signed certificate. Ensure that the private key file and the certificate file are ready.

**Replace the Certificate Files:** Add your own certificate files to the SSL certificate directory in place of the pre-existing self-signed ones. Make that the file names and formats adhere to the "easi>" system's specifications.

**Restart or Reload the program:** To effect the modifications and load the new SSL certificate, restart or reload the "easi>" program after changing the certificate files.

**Confirm the SSL setup:** Check that the new certificate is being utilized and that the SSL configuration has been updated when the program has restarted. To examine the specifics of the SSL certificate, you can use programs like OpenSSL.

**Replace the files containing certificates**. Replace the pre-existing self-signed certificate files in the SSL certificate directory with your own files. Verify that the file names and formats follow the guidelines provided by the "easi>" system.

**Reload or restart the software:** After making changes to the certificate files, restart or reload the "easi>" software to take effect of the modifications and load the new SSL certificate.

**Verify the SSL configuration:** Verify that the newly acquired certificate is being used and that the SSL configuration was updated upon program restart. You can use tools like OpenSSL to look at the details of the SSL certificate.

**Comment by ConiKost** [comment]
@john30 Any idea, if thats somehow possible?

**Comment by ConiKost** [comment, marked as answer]
Seems, that there is an option now in recent firmware versions to add an own ssl cert. Thanks!
